### PR TITLE
Libvirt vm reboot and shutdown function

### PR DESCRIPTION
**Pull request by wanghekai**
Libvirt vm reboot and shutdown function implement

### CODE REVIEW

File: piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py [hunk 1]
```diff
@@ -14,10 +17,31 @@ def __init__(self, domain_name, conn):
...
+
+    def reboot(self):
+        """
+        Reboot the VM
+        """
+        try:
+            ret = self.libvirt_domain.reboot()
+        except libvirt.libvirtError as err:
+            self.logger.error(f"API error message: {err.get_error_message()}")
+        return ret
```
Comment (Dannyb48) [inline piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py:38]: What does this actually return?

File: piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py [hunk 2]
```diff
@@ -14,10 +17,31 @@ def __init__(self, domain_name, conn):
...
+
+    def graceful_shutdown(self):
+        """
+        Gracefully shutdown the VM
+        """
+        try:
+            ret = self.libvirt_domain.destroyFlags(libvirt.VIR_DOMAIN_DESTROY_GRACEFUL)
```
Comment (Dannyb48) [inline piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py:46]: Same question?

If so can we comment what we expect to return?

File: piqe_utils/tests/vm_ops/vm_provider/libvirt/test_libvirt_vm.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+def get_domain():
+    conn = get_conn()
+    domains = conn.listDomainsID()
+    if domains:
+        return LibvirtVM(conn.lookupByID(domains[0]).name(), conn)
+    else:
+        return None
+
+
+def get_volume_params():
```
Comment (vivianQizhu) [inline piqe_utils/tests/vm_ops/vm_provider/libvirt/test_libvirt_vm.py]: You don't really need this one, right?

File: piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py (hunk 1 above)
Comment (wanghekai) [reply piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py:38]: Have already updated.Thanks

File: piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py (hunk 2 above)
Comment (wanghekai) [reply piqe_utils/api/vm_ops/vm_provider/libvirt/libvirt_vm.py:46]: Have already updated , Thanks .

File: piqe_utils/tests/vm_ops/vm_provider/libvirt/test_libvirt_vm.py (hunk 3 above)
Comment (wanghekai) [reply piqe_utils/tests/vm_ops/vm_provider/libvirt/test_libvirt_vm.py]: Correct , I have already updated , Thanks .

**Review by Dannyb48 — APPROVED** [review]
LGTM

**Pull request merged**